src/openforms/registrations/contrib/json_dump/tests/test_backend.py

```diff
@@ -0,0 +1,238 @@
+                        "url": "some://url",
+                        "name": "file1.txt",
+                        "type": "application/text",
+                        "originalName": "file1.txt",
+                    },
+                    {
+                        "url": "some://url",
+                        "name": "file2.txt",
+                        "type": "application/text",
+                        "originalName": "file2.txt",
+                    },
+                ],
+            },
+        )
+
+        SubmissionFileAttachmentFactory.create(
+            form_key="file",
+            submission_step=submission.submissionstep_set.get(),
+            file_name="file1.txt",
+            original_name="file1.txt",
+            content_type="application/text",
+            content__data=b"This is example content.",
+            _component_configuration_path="components.2",
+            _component_data_path="file",
+        )
+
+        SubmissionFileAttachmentFactory.create(
+            form_key="file",
+            submission_step=submission.submissionstep_set.get(),
+            file_name="file2.txt",
+            original_name="file2.txt",
+            content_type="application/text",
+            content__data=b"Content example is this.",
+            _component_configuration_path="components.2",
+            _component_data_path="file",
+        )
+
+        json_form_options = dict(
+            service=(ServiceFactory(api_root="http://localhost:80/")),
+            path="json_plugin",
+            variables=["file"],
+        )
+        json_plugin = JSONDumpRegistration("json_registration_plugin")
+        set_submission_reference(submission)
+
+        expected_files = [
+            {
+                "file_name": "file1.txt",
+                "content": "VGhpcyBpcyBleGFtcGxlIGNvbnRlbnQu",  # This is example content.
+            },
+            {
+                "file_name": "file2.txt",
+                "content": "Q29udGVudCBleGFtcGxlIGlzIHRoaXMu",  # Content example is this.
+            },
+        ]
+
+        res = json_plugin.register_submission(submission, json_form_options)
+
+        self.assertEqual(glom(res, "api_response.data.values.file"), expected_files)
+
+    def test_no_file_upload(self):
+        submission = SubmissionFactory.from_components(
+            [{"key": "file", "type": "file"}],
+            completed=True,
+            submitted_data={
+                "file": [],
+            },
+        )
+
+        json_form_options = dict(
+            service=(ServiceFactory(api_root="http://localhost:80/")),
+            path="json_plugin",
+            variables=["file"],
+        )
+        json_plugin = JSONDumpRegistration("json_registration_plugin")
+        set_submission_reference(submission)
+
+        res = json_plugin.register_submission(submission, json_form_options)
+
+        self.assertEqual(glom(res, "api_response.data.values.file"), {})
+
+    def test_path_traversal_attack(self):
```

Comment:
you can create some variants here and loop over them, using `self.subTest(...)` E.g. an example with `foo/../bar` or `foo..` should be blocked.